clients/src/main/java/org/apache/kafka/clients/consumer/ConsumerRebalanceListener.java

```diff
@@ -122,4 +122,30 @@
      * @throws org.apache.kafka.common.errors.InterruptException If raised from a nested call to {@link KafkaConsumer}
      */
     void onPartitionsAssigned(Collection<TopicPartition> partitions);
+
+    /**
+     * A callback method the user can implement to provide handling of cleaning up resources for partitions that have already
+     * been re-assigned to other consumers. This method will be called during normal execution as the owned partitions would
+     * first be revoked by calling the {@link ConsumerRebalanceListener#onPartitionsRevoked} first, before being re-assigned
+     * to other consumers. However, when the consumer is being kicked out of the group and hence were not aware when the new
+     * group is formed without itself, this function will then be called when the consumer finally be notified about the
+     * partitions that have already been emigrated to other consumers.
```

Comment:
I'm wondering if it would be helpful to include an example of what you might/need to do differently than onPartitionsRevoked (eg not commit offsets)

Reply:
Good idea!